File: apps/desktop/crates/console-core/src/types/git.rs
use serde::{Deserialize, Serialize};

#[derive(Clone, Debug, Serialize, Deserialize)]
#[serde(rename_all = "camelCase")]
pub struct GitFileEntry {
    pub path: String,
    pub status: String,
    pub staged: bool,
    pub additions: Option<u64>,
    pub deletions: Option<u64>,
}

#[derive(Clone, Debug, Serialize, Deserialize)]
#[serde(rename_all = "camelCase")]
pub struct GitStatusSummary {
    pub branch: String,
    pub clean: bool,
    pub files: Vec<GitFileEntry>,
}

impl GitStatusSummary {
    pub fn modified_count(&self) -> usize {
        self.files
            .iter()
            .filter(|f| f.status == "M" && !f.staged)
            .count()
    }
    pub fn staged_count(&self) -> usize {
        self.files.iter().filter(|f| f.staged).count()
    }
    pub fn untracked_count(&self) -> usize {
        self.files.iter().filter(|f| f.status == "?").count()
    }
    pub fn is_clean(&self) -> bool {
        self.clean
    }
}

#[derive(Clone, Debug, Serialize, Deserialize)]
#[serde(rename_all = "camelCase")]
pub struct GitDiffResponse {
    pub path: Option<String>,
    pub diff: String,
}

#[derive(Clone, Debug, Serialize, Deserialize)]
#[serde(rename_all = "camelCase")]
pub struct GitBranchInfo {
    pub name: String,
    pub current: bool,
}

#[derive(Clone, Debug, Serialize, Deserialize)]
#[serde(rename_all = "camelCase")]
pub struct GitBranchesResponse {
    pub branches: Vec<GitBranchInfo>,
    pub is_git_repository: bool,
}

File: apps/server-go/internal/types/git.go
// Git types ported from packages/types/src/api.ts.
package types

type GitFileStatus string // M, A, D, R, C, U, "?", "!"

type GitFileEntry struct {
	Path      string        `json:"path"`
	Status    GitFileStatus `json:"status"`
	Staged    bool          `json:"staged"`
	Additions int64         `json:"additions"`
	Deletions int64         `json:"deletions"`
}

type GitStatusSummary struct {
	Branch string         `json:"branch"`
	Clean  bool           `json:"clean"`
	Files  []GitFileEntry `json:"files"`
}

type GitBranchInfo struct {
	Name    string `json:"name"`
	Current bool   `json:"current"`
}

type GitBranchesResponse struct {
	Branches        []GitBranchInfo `json:"branches"`
	IsGitRepository bool            `json:"isGitRepository"`
}
